### PR TITLE
feat: Article about optimizing spring boot tests

### PR DESCRIPTION
**Pull request by JanCizmar**

### CODE REVIEW

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+In this article I would like to describe how I optimized our Spring Boot tests to run a bit faster.
+
+## 1. Moved from TestNG to JUnit
+I was using TestNG before, because we used it in NetSuite, where I worked before, but while using it with Spring Boot and
+kotlin I found few annoying drawbacks. 
+
+First, I wasn't able to split the tests without writing custom XML suites, where
+I would have to specify the classes. But I wanted to split the tests by its class annotation, but I couldn't find a way 
+how to do that.
```
Comment (stepan662) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:23]: Too much "but"

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+When `@DirtiesContext` annotation is used or different properties are provided for `@SpringBootTest`, the whole application
+context recreated and is similarly slow as starting of the real application. We are not using Embedded DB for integration tests
+because we want to be sure, that the code works for our production DB, and also we are running DB migration on application
+startup. Because of that and all the component scanning the app start-up takes about 15 seconds, so it's good to run tests
+which are recreating the context separately, which makes the non-recreating tests running faster, since they are not 
+interrupted by application starting.
+
+## 3. Using Bean Lazy Loading
+Since Spring Boot 2.2, we can set `spring.main.lazy-initialization` property to `true` and spring is then creating Beans 
+just, when they're needed. This makes the context creation a bit faster as well.
```
Comment (stepan662) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:42]: "just," seems weird to me

**Review by stepan662 — APPROVED** [review]

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+First, I wasn't able to split the tests without writing custom XML suites, where
+I would have to specify the classes. But I wanted to split the tests by its class annotation, but I couldn't find a way 
+how to do that.
+
+Second, I wanted to use [mockk](https://mockk.io/) and [springmockk](https://github.com/Ninja-Squad/springmockk), but 
+it's `MockkBeant` annotation wasn't working with Kotlin.
+
+Third, JUnit is used in most of the articles about testing with Spring Boot.
+
+Moving from TestNG to JUnit doesn't make the tests faster, but it was needed tto split the tests.
```
Comment (marketachalupnikova) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:30]: needed to

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+how to do that.
+
+Second, I wanted to use [mockk](https://mockk.io/) and [springmockk](https://github.com/Ninja-Squad/springmockk), but 
+it's `MockkBeant` annotation wasn't working with Kotlin.
+
+Third, JUnit is used in most of the articles about testing with Spring Boot.
+
+Moving from TestNG to JUnit doesn't make the tests faster, but it was needed tto split the tests. 
+
+## 2. Splitting up test which are recreating the Context and the rest
```
Comment (marketachalupnikova) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:32]: plural or singular? tests x is

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+When `@DirtiesContext` annotation is used or different properties are provided for `@SpringBootTest`, the whole application
+context recreated and is similarly slow as starting of the real application. We are not using Embedded DB for integration tests
+because we want to be sure, that the code works for our production DB, and also we are running DB migration on application
```
Comment (marketachalupnikova) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:35]: the application startup

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+startup. Because of that and all the component scanning the app start-up takes about 15 seconds, so it's good to run tests
```
Comment (marketachalupnikova) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:36]: scanning,

Comment (marketachalupnikova) [reply blog/2022-02-21-optimizing-spring-boot-tests.md:36]: startup or start-up? there are both versions in the text

File: blog/2022-02-21-optimizing-spring-boot-tests.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+which are recreating the context separately, which makes the non-recreating tests running faster, since they are not
```
Comment (marketachalupnikova) [inline blog/2022-02-21-optimizing-spring-boot-tests.md:37]: run faster

**Pull request merged**